Add comment checklist segment (#15)
makerdao/pe-checklists: spell/spell-crafter-goerli-workflow.md

Reviewer: Please don't forget to apply comments to every other file

  * [ ] Set `deployed_spell` to `address(0)`
  * [ ] Set `deployed_spell_created` to `0`
  * [ ] Set `deployed_spell_block` to `0`

> * [ ] Add comments to the spell:

Reviewer: Sorry, let's move it one more section down under
```
* [ ] Cleanup Specific Tests in `DssSpell.t.sol`
```
and before
```
* [ ] Run Tests `make test` or `make test match=<test_name>` to inspect debug traces
```

  * [ ] Copy every _Section text_ from the Exec Sheet as comment to the spell code (above the code segment that implements the action)
    * [ ] Surround the comment by the set of dashes (E.g. `// ----- Section text -----`)
  * [ ] For every _Instruction text_ Exec Sheet:
    * [ ] copy to the spell code as `// Instruction text`
    * [ ] add newline above it
  * [ ] For every `Reasoning URL` and `Authority URL` from the Exec Sheet add comment with it under relevant section or instruction in the spell code (depending on which row the link is present)
    * [ ] For every `Reasoning URL` and `Authority URL` add prefix derived from the url itself

>       * [ ] `// Executive Vote:` if URL starts with `https://vote.makerdao.com/executive/`
>       * [ ] `Poll:` if URL starts with `https://vote.makerdao.com/polling/`
>       * [ ] `Forum:` if URL starts with `https://forum.makerdao.com/t/`
>       * [ ] `MIP:` if URL starts with `https://mips.makerdao.com/mips/details/`

Reviewer: all other URL types were not yet fixed

```suggestion
      * [ ] `// Executive Vote:` if URL starts with `https://vote.makerdao.com/executive/`
      * [ ] `// Poll:` if URL starts with `https://vote.makerdao.com/polling/`
      * [ ] `// Forum:` if URL starts with `https://forum.makerdao.com/t/`
      * [ ] `// MIP:` if URL starts with `https://mips.makerdao.com/mips/details/`
```

  * [ ] Use DssExecLib Functions
    * [ ] Ensure `DssExecLib` address used in current spell (`DssExecLib.address`) matches `dss-exec-lib` [Latest Release Tag](https://github.com/makerdao/dss-exec-lib/releases/latest)
    * [ ] Check previous spells for common patterns

> * [ ] Add comments to the spell:

Reviewer: I would actually move it much higher, right after:
```
* [ ] Cleanup `src/test/config.sol`
```
(same for the reviewers checklist)

The reason are:
- Content sections are useful for the cleanup especially because they allow conflict-free contributions from third parties
- Review should always start with checking if instructions are correctly transferred to the code and then use them as the reference for the actual code. Because if instructions are incorrect (or outdated) then the code is most probably too

  * [ ] Copy every _Section text_ from the Exec Sheet as comment to the spell code (above the code segment that implements the action)

>     * [ ] Surround the comment by the set of dashes (E.g. `----- Section text -----`)

Reviewer: ```suggestion
    * [ ] Surround the comment by the set of dashes (E.g. `// ----- Section text -----`)
```

  * [ ] For every _Instruction text_ Exec Sheet:
    * [ ] copy to the spell code as `// Instruction text`
    * [ ] add newline above it
  * [ ] For every `Reasoning URL` and `Authority URL` from the Exec Sheet add comment with it under relevant section or instruction in the spell code (depending on which row the link is present)
    * [ ] For every `Reasoning URL` and `Authority URL` add prefix derived from the url itself

>       * [ ] `Executive Vote:` if URL starts with `https://vote.makerdao.com/executive/`

Reviewer: ```suggestion
      * [ ] `// Executive Vote:` if URL starts with `https://vote.makerdao.com/executive/`
```
same for all other URL types

      * [ ] `Poll:` if URL starts with `https://vote.makerdao.com/polling/`
      * [ ] `Forum:` if URL starts with `https://forum.makerdao.com/t/`
      * [ ] `MIP:` if URL starts with `https://mips.makerdao.com/mips/details/`
  * [ ] If action in the spell doesn't have relevant instruction (e.g.: `chainlog` version bump), it add explanation prefixed with `// Note:`

>   * [ ] If an instruction can not be taken, a note add under the instruction prefixed with `// Note: ` (e.g.: `// Note: Skip payments on goerli`)

Reviewer: ```suggestion
  * [ ] If an instruction can not be taken, add a comment under the instruction prefixed with `// Note:` (e.g.: `// Note: Payments are skipped on goerli`)
```
